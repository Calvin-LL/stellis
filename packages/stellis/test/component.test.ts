import { describe, it, expect } from 'vitest';
import transformStellis from './utils';

describe('component', () => {
  it('should transform basic', async () => {
    const code = `
      <Heading>Hello World</Heading>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
  it('should transform with static attribute', async () => {
    const code = `
      <Heading title="hello world">Hello World</Heading>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
  it('should transform with dynamic attribute', async () => {
    const code = `
      <Heading title={getTitle()}>Hello World</Heading>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
  it('should transform with async dynamic attribute', async () => {
    const code = `
      <Heading title={await getTitle()}>Hello World</Heading>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
  it('should transform with static child element', async () => {
    const code = `
      <Heading>
        <h1>Hello World</h1>
        <h1>Hello World</h1>
      </Heading>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
  it('should transform with dynamic child element', async () => {
    const code = `
      <Heading>
        {dynamicChild()}
        {dynamicChild()}
      </Heading>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
  it('should transform with async dynamic child element', async () => {
    const code = `
      <Heading>
        {await dynamicChild()}
        {await dynamicChild()}
      </Heading>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
});
