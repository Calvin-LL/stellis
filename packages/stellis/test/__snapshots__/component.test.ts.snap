// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component > should transform basic 1`] = `
"import { $$component as _$$component } from \\"stellis\\";
_$$component(Heading, {
  children: \\"Hello World\\"
});"
`;

exports[`component > should transform with async dynamic attribute 1`] = `
"import { $$component as _$$component } from \\"stellis\\";
async () => {
  const _v = await getTitle();
  return _$$component(Heading, {
    title: _v,
    children: \\"Hello World\\"
  });
};"
`;

exports[`component > should transform with async dynamic child element 1`] = `
"import { $$component as _$$component } from \\"stellis\\";
_$$component(Heading, {
  children: [async () => await dynamicChild(), async () => await dynamicChild()]
});"
`;

exports[`component > should transform with dynamic attribute 1`] = `
"import { $$component as _$$component } from \\"stellis\\";
_$$component(Heading, {
  title: getTitle(),
  children: \\"Hello World\\"
});"
`;

exports[`component > should transform with dynamic child element 1`] = `
"import { $$component as _$$component } from \\"stellis\\";
_$$component(Heading, {
  children: () => [dynamicChild(), dynamicChild()]
});"
`;

exports[`component > should transform with static attribute 1`] = `
"import { $$component as _$$component } from \\"stellis\\";
_$$component(Heading, {
  title: \\"hello world\\",
  children: \\"Hello World\\"
});"
`;

exports[`component > should transform with static child element 1`] = `
"import { $$html as _$$html } from \\"stellis\\";
import { $$component as _$$component } from \\"stellis\\";
const _template = \\"<h1>Hello World</h1>\\",
  _template2 = \\"<h1>Hello World</h1>\\";
_$$component(Heading, {
  children: [_$$html(_template, true), _$$html(_template2, true)]
});"
`;
